components/ChallengeMaterial.tsx

```diff
@@ -312,8 +320,8 @@ const ChallengeMaterial: React.FC<ChallengeMaterialProps> = ({
   }
   //create an object to evaluate the student's status with a challenge
   const userSubmissionsObject: UserSubmissionsObject = userSubmissions.reduce(
-    (acc: UserSubmissionsObject, submission: UserSubmission) => {
-      acc[submission.challengeId] = submission
+    (acc: UserSubmissionsObject, submission: Submission) => {
```

Comment:
The type here is inferred from the array type so you don't need to specify it.

```suggestion
    (acc: UserSubmissionsObject, submission) => {
```